### ISSUE TITLE
[99.15] Insert Link - Node - search only finds nodes already open in the tree.  

### ISSUE THREAD

**Issue by dbwiz** [giuspen/cherrytree#1246]
**Version**
Windows 10 1909.
Portable installer: cherrytree_0.99.15.0_win64_portable.7z

**Description**
When creating a link to a node in the "Insert/Edit Link" form you were able to type the starting letters of the name of the node and the node would be found and selectable in the tree.  The tree in the form is expanded the same as the tree on the main screen.  In 99.15, the node is only found if it is visible in the tree.  In 39.4 and previous, the node is found whether it's visible or not.  

**To Reproduce**
1. Open with copy of existing notebook edited with 39.4
2. Get the name of a uniquely named node and collapse the node so it isn't visible in the tree.
3. Select a node and a word and choose "insert Link" "node"
4. Start to type the name of the uniquely named node that isn't visible (i.e. "Oct").   In 39.4, it finds the named node but in 99.15 it doesn't.
5. Exit "insert link" and make that node visible in the tree
6. Reattempt to insert the link as in 3.  After clicking the "Node" radio button, start typing the name of the unique node and it will be found.

**Impact**
This is an important feature for me.  I extensively link my notebook which has added several hundred nodes this year.  It's tedious to navigate through the tree to find the node I need to link to, and I don't really want to have the whole tree expanded all the time.

Thank you so much for writing CherryTree.  I've tried many different paper and digital systems (including onenote) in the last 30 years and CherryTree is the one that fits the best, and I've been using it for 5 years now.  My notebook is constantly open and updated all day long and it's given everyone the illusion that I'm super organized.  Thank you.

**updated for clarity**

**Comment by dbwiz** [comment]
Note there is a another issue that I'll report here as it seems to be related.  If you would like this as a separate report please let me know and I'll issue it.
I open a copy of my main notebook (200MB .ctb) vaccumed with 39.4.
When I hit "F8", it presents the "waiting ring" but does not create or go to the current date page (wait for up to 5 mins then I kill the session)
If I manually create the current date node as F8 would, leave it visible in the tree, save and exit the notebook and go back in and hit F8 - it finds the node.
If I then save the notebook with the current date node hidden, exit and go back in and hit F8 - it then presents the same "waiting ring" issue.
Thus, the F8 code locks up if the current date node isn't visible.

**Comment by giuspen** [comment]
I'm actually surprised that in the old pygtk version simply typing would select a node that is not visible in the tree, but I'll try soon and get back

**Comment by dbwiz** [comment]
Apologies, I've retested this and while I thought 39.4 found items in hidden nodes, it does not.  Not sure how I made that mistake, I was quite certain it worked that way.  After looking at how I use CT I think almost all of the time I create a new node and leave it open, then copy the properties to the page I want to link it to (often the daily/monthly page).  I think my primary nodes with lists of issues etc are always open due to constantly referring to them.  So linking to hidden nodes is not so useful in practice.    Thank you - and thank you for CT!
